README.md

```diff
@@ -1,34 +1,115 @@
+
+  - This ViewModel interacts with the `TripRepository` to fetch, create, update, and delete trips. It
+  - Handles the management of trip-related states such as the selected trip, selected day, activities, and UI state.
+  - It's responsible for handling file uploads to Firebase Storage and integrating with the AI assistant to generate trip activities.
+
+- PlacesViewModel:
+  - ViewModel for managing places-related data and operations in the app.
+  - This ViewModel interacts with the `PlacesRepository` to fetch places, search for places, and get place details.
+  - It handles the management of place-related states such as the selected place, search results, and UI state.
+
+##### Repositories:
+
+- Interfaces:
+
+  - `UserRepository`: Interface for managing user (basic CRUD operations) interactions with a data source.
+  - `TripRepository`: Interface for managing trip (basic CRUD operations) interactions with a data source.
+  - `PlacesRepository`: Interface for managing places-related interactions with a location (map-based) data source.
+  - `FriendRequestRepository` and `TripInviteRepository`: used managing friend requests and trip invites.
+
+- Repositproies (implementing the interfaces for Firestore and Google Maps API use cases):
+  - `UserRepositoryFirebase`
+  - `TripRepositoryFirebase`
+  - `GooglePlacesRepository`
+  - `FriendRequestRepositoryFirebase` and `TripInviteRepositoryFirebase`
+
+##### Screens:
+
+Figma design visible [here](https://www.figma.com/design/TZ4qU0PNmMv3T6EpesSrL9/Voyageur?node-id=79-2&p=f) includes these main screens / composables:
+
+- `SignIn` screen: Allows users to sign in using Google (Firebase) authentication without the need to enter any additional details.
+- `Overview` screen: Displays a list of trips and allows users, a search bar to filter for trips, and a floating action button to create a new trip. It is also possible to add trips to favourites.
+- `Profile` screen: Displays user information, allows users to edit their profile, and view, accept or reject their friend requests and trips invites. Also allows users to manage their interests that may be used by the AI assistant.
+- `AddTrip` and `Settings` (for a trip) screens: Allow users to create/edit a trip and configure settings (participants).
+- `Search` screen: Allows users to search for other users and places based on location. It also displays a feed of public trips.
+- Trip screens:
+  - `WeeklyView` / `ByDay` screens: Display the trip itinerary in a weekly view or in a certain day, allowing users to view and edit activities. It also allows users to generate a schedule for a day using the AI assitant.
+  - `Map` screen: Displays a map with markers for places and activities, allowing users to view details in a trip.
+  - `Phtos` screen: Displays photos shared by participants in a trip.
+  - `Activities` screen: Displays a list of activities in a trip, allowing users to view and edit details.
```

Comment:
There's also the _ActivitiesForOneDay_, the _ActivitiesMap_ Screen and the _Assistant_ Screen. We should also mention these in the README for the app.

Reply:
`ActivitiesMap` is `Map screen`. I will mention the 2 others, thanks for catching this.